### ISSUE TITLE
Login fails with HTTPS

### ISSUE THREAD

**Issue by bialy03** [datenstrom/yellow#434]
When my site has HTTPS enabled, the login form does not want to log in to me.
Please, help me.

**Comment by schulle4u** [comment]
Yes, I can confirm a problem for SSL-enabled sites. If you use https to login and the next time try to access your edit URL using http, you cannot login for some reason. Not sure if this is a bug, but there's an easy workaround for the problem. Try enabling the HSTS option for your domain so that all requests automatically use https.

**Comment by bialy03** [comment]
I turned on HSTS and thanks to that I did not have a problem with the installation, but the problem with logging still occurs

**Comment by schulle4u** [comment]
Have you tried deleting the browser cookies for Yellow? Maybe an old cookie prevents you from logging into your site. 
Please give some more detailed steps to help us reproduce the problem.

**Comment by bialy03** [comment]
Since I turned on HSTS, the site automatically returned to HTTP. I removed the cookies, but it did not help.

**Comment by markseu** [comment]
Can you give more details, @bialy03. Does the login fail in the official demo as well? What happens if you use a new/unmodified Datenstrom Yellow on your server? What software do you use on your server, please copy/paste your file `system/extensions/yellow.log`.

**Comment by bialy03** [comment]
```
2019-06-27 10:44:33 info Datenstrom Yellow 0.8.5, PHP 7.1.26, Apache Linux
2019-06-27 10:44:33 info Check Apache server configuration
2019-06-27 10:44:33 info Install language 'English'
2019-06-27 10:44:33 info Install language 'German'
2019-06-27 10:44:33 info Install language 'Polish'
2019-06-27 10:44:51 info Install extension 'Blog 0.8.4'
2019-06-27 10:44:51 info Add user 'LSO'
2019-06-30 09:23:47 info Update extension 'English 0.8.12'
2019-06-30 09:23:47 info Update extension 'German 0.8.12'
2019-06-30 09:23:47 info Update extension 'Polish 0.8.12'
```
I have free shared hosting and I do not know much about the software on their server.
PS On which server is the Yellow demo set up? Maybe I'll move there.

**Comment by markseu** [comment]
The official demo is here https://extensions.datenstrom.se/ - please try the login here.

The server is hosted at https://www.servado.de, very reliable.

**Comment by bialy03** [comment]
I do not have any problems logging in on the official demo.

wt., 2 lip 2019, 13:31 użytkownik Mark Seuffert <notifications@github.com>
napisał:

**Comment by markseu** [comment]
Thanks for the information.

I can't reproduce the described problems on my Linux server. As far as I can see from the logfile, it should work on your server too. Not sure what's the best way to narrow down the problem.  

If you want me to investigate, please [contact me](https://datenstrom.se/contact/) and give me access to your server.

**Comment by bialy03** [comment]
It seems to me that it depends on connecting the domain and the certificate.

wt., 2 lip 2019, 13:50 użytkownik Mark Seuffert <notifications@github.com>
napisał:

**Comment by IngoBelka** [comment]
Only a thought. I had a similar problem on my system. Shared webspace often is situated behind a load balancer. Mostly the environment variable "X-Forwarded-Proto: https” ist not forwarded to the script. I had a system driven by PERL and I could set the following in my config file: $ENV{HTTPS} = 'ON'; to enforce https. May be it's possible to introduce a similar setting for yellow?

**Comment by markseu** [comment]
Good thought. There's a setting for that, you can use `ServerUrl` in in file `system/config/config.ini`. Datenstrom Yellow will usually detect the server URL, but you can also pin down a specific URL.

**Comment by markseu** [comment]
Good thought. There's a setting for that, you can use `ServerUrl` in file `system/settings/system.ini`. Datenstrom Yellow will usually detect the server URL, but you can also pin down a specific URL.

**Comment by datenstromse** [comment]
Closing issue, there is not enough information to reproduce a problem at the moment.

**Comment by neto737** [comment]
I had the same issue today. I'm trying to install it in a domain .dev (that requires HSTS), and I'm unable to login in /edit/.

Trying to solve the problem and after seeing this issue, I've just installed the script in another domain without SSL and I was able to login. In the same domain without SSL that I was able to login, after enabling SSL I got unable to login again.

Anyway, I can provide you all details or even a complete subdomain with and without SSL enabled to make your tests and may fix it. @datenstromse
